proving/proving_test.go

```diff
@@ -5,65 +5,73 @@ import (
 	"encoding/binary"
 	"flag"
 	"fmt"
-	"io/ioutil"
+	"log"
 	"math"
-	"os"
 	"testing"
 
 	"github.com/spacemeshos/post/config"
 	"github.com/spacemeshos/post/initialization"
 	"github.com/spacemeshos/post/shared"
 	"github.com/spacemeshos/post/verifying"
-	smlog "github.com/spacemeshos/smutil/log"
 	"github.com/stretchr/testify/require"
 )
 
 var (
 	id = make([]byte, 32)
 	ch = make(Challenge, 32)
 
-	cfg  config.Config
-	opts config.InitOpts
-
-	log   = flag.Bool("log", false, "")
-	debug = flag.Bool("debug", false, "")
+	logFlag   = flag.Bool("log", false, "")
+	debugFlag = flag.Bool("debug", false, "")
 
 	NewInitializer = initialization.NewInitializer
 	CPUProviderID  = initialization.CPUProviderID
 )
 
-func TestMain(m *testing.M) {
-	cfg = config.DefaultConfig()
+func getTestConfig(t *testing.T) (config.Config, config.InitOpts) {
+	cfg := config.DefaultConfig()
 	cfg.LabelsPerUnit = 1 << 12
 
-	opts = config.DefaultInitOpts()
-	opts.DataDir, _ = ioutil.TempDir("", "post-test")
+	opts := config.DefaultInitOpts()
+	opts.DataDir = t.TempDir()
 	opts.NumUnits = cfg.MinNumUnits
 	opts.NumFiles = 2
 	opts.ComputeProviderID = CPUProviderID()
 
-	res := m.Run()
-	os.Exit(res)
+	return cfg, opts
+}
+
+type testLogger struct {
+	shared.Logger
+}
+
+func (l testLogger) Info(msg string, args ...interface{}) {
+	if *logFlag {
+		log.Printf("\tINFO\t"+msg, args...)
+	}
+}
+
+func (l testLogger) Debug(msg string, args ...interface{}) {
+	if *debugFlag {
+		log.Printf("\tDEBUG\t"+msg, args...)
+	}
 }
```

Comment:
If you stick with this, then I guess it deserves to be factored out to some `test_helpers` pkg.

Reply:
I am strongly against this. With this PR I actually did the exactly the opposite; I got rid of the `test_helpers` pkg (called `smutil`) because this is code that doesn't need to be shared.

There is nothing wrong with having your own implementation of the `shared.Logger` interface in every package. It just happens to be the same here as in the other package.

The https://github.com/spacemeshos/smutil package now serves no purpose any more, afaik no one is using it any more. If you look into it it contains 1:1 the code we JUST removed from the `filesystem` package in go-spacemesh. This is what happens to code that is factored out for re-usage: it is used in exactly one place and forgotten about.

Reply:
The logic is so trivial it doesn't deserve to be it's own package (with documentation of its API and so on!).

If it helps see `testLogger` as a mock of the `shared.Logger` interface. Now pulling in `gomock` is more effort than having this small handwritten mock.

Reply:
It's not the kind of code you would expect to be different in each pkg. I would actually expect to have consistent logging in tests in various pkgs.

Reply:
I would say otherwise. The interface of the logger isn't set by the tests, it's set by the post `Initializer`.

So the package defining that interface would need to provide an implementation for tests. How do you then use the same logger for two different components with different interfaces?